filament/src/details/Material.cpp

```diff
@@ -220,17 +220,9 @@ FMaterial::FMaterial(FEngine& engine, const Material::Builder& builder)
             mRasterState.blendFunctionDstAlpha = BlendFunction::ZERO;
             mRasterState.depthWrite = true;
             break;
+        // Do not change this behavior without checking for regressions with AlphaBlendModeTest
+        // and TextureLinearInterpolationTest, with and without View::BlendMode::TRANSLUCENT.
         case BlendingMode::MASKED:
```

Comment:
I don't like this fix because if your GPU is not a tiler then there is 0 benefit to using alpha masking over transparent blending. It would make rendering vegetations on desktop for instance incredibly expensive.

In addition you removed `depthWrite = true`, it needs to be brought back.

The blend functions should be the same as `OPAQUE` when the render target is opaque, and the same as `TRANSPARENT` (+ `depthWrite=true`) when the render target is transparent.

Reply:
Good catch on the `depthWrite` line, fixed.